code/game/machinery/ecmo.dm

```diff
@@ -0,0 +1,594 @@
+/obj/machinery/ecmo
+	name = "\improper ECMO"
+	desc = "An Extra-Corporeal Membrane Oxygenator."
+	desc_info = "ECMO must be supplied with a blood pack and an oxygen tank."
+	icon = 'icons/obj/ecmo.dmi'
+	icon_state = "ecmo_stand"
+	anchored = 0
+	density = FALSE
+	var/tipped = FALSE
+	var/last_full // Spam check
+	var/last_warning
+
+	// Blood Stuff
+	var/mob/living/carbon/human/attached = null
```

Comment:
See below, but I advise a single `patient` variable for the entire machine.

Reply:
I don't think this would work, when each cannula is inserted the appropriate variable is set with the mob that has it and is subsequently checked in each process if the correct cannula is inserted (and thus the variable set), having a single variable would mean either that:

- Either it does everything or nothing
- Another variable has to be created to know the state of one of the two cannulas, thus defeating the purpose of two cannulas